components/image-builder-mk3/pkg/orchestrator/orchestrator.go

```diff
@@ -334,8 +336,15 @@ func (o *Orchestrator) Build(req *protocol.BuildRequest, resp protocol.ImageBuil
 		bobBaseref += ":latest"
 	}
 	wsref, err := reference.ParseNamed(wsrefstr)
+	var workspaceAuth []byte
 	var additionalAuth []byte
 	if err == nil {
+		res := make(auth.ImageBuildAuth)
+		res[reference.Domain(wsref)] = registry.AuthConfig(*wsrefAuth)
+		workspaceAuth, err = json.Marshal(map[string]auth.ImageBuildAuth{"auths": res})
```

Comment:
By passing workspace auth explicitly here, the additional auth no longer becomes necessary, i.e. we can just merge the two. The only reason they were separate before was because the workspace author originally came through with the secret.

Reply:
So as per my other PR #18604 It looks like the strategy I've settled on is to still pass through an amalgamated auth in the additionalAuth field where it can be used by the proxy and builder, whilst the workspaceAuth contains just the auth needed for pulling the wsref directly. I'm happy to still combine them into a single var, but it actually works out more elegant keeping them separate.